pkg/bpmn/flow_node_executor.go

```diff
@@ -0,0 +1,86 @@
+package bpmn
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pbinitiative/zenbpm/pkg/bpmn/model/bpmn20"
+	"github.com/pbinitiative/zenbpm/pkg/bpmn/runtime"
+	"github.com/pbinitiative/zenbpm/pkg/storage"
+)
+
+// FlowNodeExecutor defines the behavior of all BPMN executors
+type FlowNodeInstanceExecutor interface {
+	Execute(ctx context.Context, batch storage.Batch, engine *Engine, process *runtime.ProcessDefinition, instance *runtime.ProcessInstance, originActivity runtime.Activity) (createFlowTransitions bool, activityResult runtime.Activity, nextCommands []command, err error)
+}
+
+func GetExecutorDefinition(elementType bpmn20.ElementType) func(elementActivity) FlowNodeInstanceExecutor {
```

Comment:
returning a constructor function seems too complicated to me.
Why not simply create the the executor straight, which reduces lines of code by half and is more readable... what do you think?
Example, what I mean:
```suggestion
func createFlowNodeInstanceExecutor(elementType bpmn20.ElementType, ea *elementActivity) FlowNodeInstanceExecutor {
	switch elementType {
	// Activities
	case bpmn20.ElementTypeServiceTask:
		return &ServiceTaskExecutor{ActivityExecutor{FlowNodeExecutor{flowNode: ea}}}
[...]
```